data/collector.go

```diff
@@ -68,23 +69,26 @@ func (c *collector) Close() {
 }
 
 // Collect starts the Collector, causing it to run c.capturer.Capture every c.interval, and write the results to
-// c.target.
+// c.target. It blocks until the underlying capture goroutine starts.
 func (c *collector) Collect() {
 	_, span := trace.StartSpan(c.cancelCtx, "data::collector::Collect")
 	defer span.End()
 
+	started := make(chan struct{})
 	c.backgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
 		defer c.backgroundWorkers.Done()
+		close(started)
 		c.capture()
 	})
 	c.backgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
 		defer c.backgroundWorkers.Done()
 		if err := c.write(); err != nil {
-			c.logger.Errorw(fmt.Sprintf("failed to write to collector %s", c.target.Directory), "error", err)
+			c.logger.Errorw("failed to write to collector", "error", err)
```

Comment:
Still seems useful to include the target directory

Reply:
Added back